ls1intum/Artemis
# `Programming exercises`: Allow starting practice mode from the last graded version

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


When the student wants to practice after the due date, they should have the option to either start from scratch or keep going from their graded participation if they did not get 100% there.

### Description

The student can now choose to start from the graded participation when starting the practice mode.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 2 Programming Exercises (one with a participation before the duedate one without)

1. Log in to Artemis
2. Participate before the deadline in the one exercise
3. Check that you can start the practice mode after the deadline in both exercises and have the option to use the graded participation in one of them

### Review Progress





#### Code Review
#### Manual Tests
#### Exam Mode Test
- [ ] Test 1 (Should have the same UI and behavior as before)

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| start-practice-mode-button.component.ts | 77% | 90% |
| course-exercise.service.ts | 72% | 96% |
| ParticipationService.java | 78% | 92% |


### Screenshots


![Bildschirmfoto 2022-10-13 um 11 53 30](https://user-images.githubusercontent.com/38322605/195565524-28431298-6a17-4fc4-9297-6fcc60190667.png)
